### ISSUE TITLE
fly cask has unsigned binary; needs quarantine xattr deleted

### ISSUE THREAD

**Issue by todd-a-jacobs** [Homebrew/homebrew-cask#98458]
#### General troubleshooting steps

- [x] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [x] I have retried my command with `--force`.
- [x] I ran `brew update-reset && brew update` and retried my command.
- [x] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [x] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [x] I made doubly sure this is not a [checksum does not match](https://github.com/Homebrew/homebrew-cask/blob/master/doc/reporting_bugs/checksum_does_not_match_error.md) error.

#### Description of issue

On Big Sur and testing on multiple machines, running the golang binary installed by the `fly` cask is not permitted without taking additional steps because it's unsigned. While the user can force the binary to run via the GUI, this can also be done using the *xattr* command as outlined below:

``` bash
xattr -dr com.apple.quarantine "$(brew --prefix)/bin/fly"
```

This (or something functionally equivalent) should be added to the cask's recipe.

#### Command that failed

```
brew install --cask fly

# these will trigger an unidentified developer warning; see image below
fly
fly --version
```

<img width="372" alt="Screen Shot 2021-01-20 at 6 01 39 PM" src="https://user-images.githubusercontent.com/385462/105252486-81020480-5b4b-11eb-9961-81ec5139c3fd.png">

#### Output of command with `--verbose --debug`

```
$ brew install --cask fly --verbose --debug
==> Cask::Installer#install
==> Printing caveats
==> Cask::Installer#fetch
==> Downloading https://github.com/concourse/concourse/releases/download/v6.7.2/fly-6.7.2-darwin-amd64.tgz
/usr/bin/curl --disable --globoff --show-error --user-agent Homebrew/2.7.5-10-g8869208\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 11.1\)\ curl/7.64.1 --header Accept-Language:\ en --retry 3 --location --silent --head --request GET https://github.com/concourse/concourse/releases/download/v6.7.2/fly-6.7.2-darwin-amd64.tgz
Already downloaded: /Users/foo/Library/Caches/Homebrew/downloads/6c6d235c6d65e0238d0cf764c2dc3c0719a5e7e82ad316a5344a5f0b5be1b1aa--fly-6.7.2-darwin-amd64.tgz
==> Checking quarantine support
/usr/bin/xattr
/usr/bin/swift /usr/local/Homebrew/Library/Homebrew/cask/utils/quarantine.swift
==> Quarantine is available.
==> Verifying Gatekeeper status of /Users/foo/Library/Caches/Homebrew/downloads/6c6d235c6d65e0238d0cf764c2dc3c0719a5e7e82ad316a5344a5f0b5be1b1aa--fly-6.7.2-darwin-amd64.tgz
/usr/bin/xattr -p com.apple.quarantine /Users/foo/Library/Caches/Homebrew/downloads/6c6d235c6d65e0238d0cf764c2dc3c0719a5e7e82ad316a5344a5f0b5be1b1aa--fly-6.7.2-darwin-amd64.tgz
==> /Users/foo/Library/Caches/Homebrew/downloads/6c6d235c6d65e0238d0cf764c2dc3c0719a5e7e82ad316a5344a5f0b5be1b1aa--fly-6.7.2-darwin-amd64.tgz is quarantined
==> Verifying checksum for cask 'fly'.
tar tf /Users/foo/Library/Caches/Homebrew/downloads/6c6d235c6d65e0238d0cf764c2dc3c0719a5e7e82ad316a5344a5f0b5be1b1aa--fly-6.7.2-darwin-amd64.tgz
==> Installing Cask fly
==> Cask::Installer#stage
==> Extracting primary container
==> Using container class UnpackStrategy::Tar for /Users/foo/Library/Caches/Homebrew/downloads/6c6d235c6d65e0238d0cf764c2dc3c0719a5e7e82ad316a5344a5f0b5be1b1aa--fly-6.7.2-darwin-amd64.tgz
tar xof /Users/foo/Library/Caches/Homebrew/downloads/6c6d235c6d65e0238d0cf764c2dc3c0719a5e7e82ad316a5344a5f0b5be1b1aa--fly-6.7.2-darwin-amd64.tgz -C /var/folders/tw/dtbqztd57p5_z9c85dk5znvw0000gn/T/d20210120-54654-11djhxu
chmod +rw /var/folders/tw/dtbqztd57p5_z9c85dk5znvw0000gn/T/d20210120-54654-11djhxu/fly
hdiutil imageinfo -format /var/folders/tw/dtbqztd57p5_z9c85dk5znvw0000gn/T/d20210120-54654-11djhxu/fly
cp -p /var/folders/tw/dtbqztd57p5_z9c85dk5znvw0000gn/T/d20210120-54654-11djhxu/fly /usr/local/Caskroom/fly/6.7.2/fly
==> Verifying Gatekeeper status of /Users/foo/Library/Caches/Homebrew/downloads/6c6d235c6d65e0238d0cf764c2dc3c0719a5e7e82ad316a5344a5f0b5be1b1aa--fly-6.7.2-darwin-amd64.tgz
/usr/bin/xattr -p com.apple.quarantine /Users/foo/Library/Caches/Homebrew/downloads/6c6d235c6d65e0238d0cf764c2dc3c0719a5e7e82ad316a5344a5f0b5be1b1aa--fly-6.7.2-darwin-amd64.tgz
==> /Users/foo/Library/Caches/Homebrew/downloads/6c6d235c6d65e0238d0cf764c2dc3c0719a5e7e82ad316a5344a5f0b5be1b1aa--fly-6.7.2-darwin-amd64.tgz is quarantined
==> Propagating quarantine from /Users/foo/Library/Caches/Homebrew/downloads/6c6d235c6d65e0238d0cf764c2dc3c0719a5e7e82ad316a5344a5f0b5be1b1aa--fly-6.7.2-darwin-amd64.tgz to /usr/local/Caskroom/fly/6.7.2
/usr/bin/xattr -p com.apple.quarantine /Users/foo/Library/Caches/Homebrew/downloads/6c6d235c6d65e0238d0cf764c2dc3c0719a5e7e82ad316a5344a5f0b5be1b1aa--fly-6.7.2-darwin-amd64.tgz
/usr/bin/xargs -0 -- /bin/chmod -h u\+w
/usr/bin/xargs -0 -- /usr/bin/xattr -w com.apple.quarantine 0181\;6005e7f7\;Homebrew\\x20Cask\;F6F15DAC-53E4-4D84-B240-05DED4680F0A
==> Creating metadata directory /usr/local/Caskroom/fly/.metadata/6.7.2/20210120230659.423.
==> Creating metadata subdirectory /usr/local/Caskroom/fly/.metadata/6.7.2/20210120230659.423/Casks.
==> Installing artifacts
==> 1 artifact/s defined
#<SortedSet: {#<Cask::Artifact::Binary:0x00007fbb031b5880 @cask=#<Cask::Cask:0x00007fbb1033c368 @token="fly", @sourcefile_path=#<Pathname:/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/fly.rb>, @tap=#<Tap:0x00007fbb031956c0 @user="Homebrew", @repo="cask", @name="homebrew/cask", @full_name="Homebrew/homebrew-cask", @path=#<Pathname:/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask>, @alias_table=nil, @alias_reverse_table=nil, @formula_renames={}, @cask_dir=#<Pathname:/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks>>, @block=#<Proc:0x00007fbb1033c2c8@/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/fly.rb:1>, @default_config=#<Cask::Config:0x00007fbb1033f5e0 @explicit={}, @binarydir=#<Pathname:/usr/local/bin>>, @caskroom_path=#<Pathname:/usr/local/Caskroom/fly>, @metadata_master_container_path=#<Pathname:/usr/local/Caskroom/fly/.metadata>, @config=#<Cask::Config:0x00007fbb031b6b90 @explicit={}, @binarydir=#<Pathname:/usr/local/bin>>, @dsl=#<Cask::DSL:0x00007fbb031b6960 @cask=#<Cask::Cask:0x00007fbb1033c368 ...>, @token="fly", @version="6.7.2", @sha256=#<Checksum:0x00007fbb031b6730 @hexdigest="e095e0d631cd42a28b63e7101819840bad1b255d8b6d31ceb1fc0220c055605d">, @url=#<URL:0x00007fbb031b62d0 @uri=#<URI::HTTPS https://github.com/concourse/concourse/releases/download/v6.7.2/fly-6.7.2-darwin-amd64.tgz>, @verified=nil, @using=nil, @tag=nil, @branch=nil, @revisions=nil, @revision=nil, @trust_cert=nil, @cookies=nil, @referer=nil, @header=nil, @user_agent=:default, @data=nil, @specs={:user_agent=>:default}, @from_block=false, @caller_location="/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/fly.rb:5:in `block in load'">, @appcast=#<Cask::DSL::Appcast:0x00007fbb031b5f60 @uri=#<URI::HTTPS https://github.com/concourse/concourse/releases.atom>, @parameters={}>, @name=["fly"], @desc="Official CLI tool for Concourse CI", @homepage="https://github.com/concourse/concourse", @artifacts=#<SortedSet: {...}>, @staged_path=#<Pathname:/usr/local/Caskroom/fly/6.7.2>, @language_eval=nil>>, @source_string="fly", @target_string="", @source=#<Pathname:/usr/local/Caskroom/fly/6.7.2/fly>, @target=#<Pathname:/usr/local/bin/fly>>}>
==> Installing artifact of class Cask::Artifact::Binary
==> Linking Binary 'fly' to '/usr/local/bin/fly'.
/bin/ln -h -f -s -- /usr/local/Caskroom/fly/6.7.2/fly /usr/local/bin/fly
/usr/local/Homebrew/Library/Homebrew/shims/scm/git --version
🍺  fly was successfully installed!
```

#### Output of `brew doctor --verbose`

```
==> Cask Environment Variables:
BUNDLE_PATH
CHRUBY_VERSION
GEM_HOME
GEM_PATH
HOMEBREW_CASK_OPTS
LC_ALL
PATH
RBENV_VERSION
RUBYLIB
RUBYOPT
RUBYPATH
SHELL
==> $LOAD_PATHS
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/spoom-1.0.7/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/thor-1.0.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/sorbet-runtime-stub-0.2.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ruby-macho-2.5.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-sorbet-0.5.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-rspec-2.1.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-rails-2.9.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-performance-1.9.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ruby-progressbar-1.11.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-wait-0.0.9/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-sorbet-1.8.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-retry-0.6.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-its-1.3.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-3.10.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-mocks-3.10.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-expectations-3.10.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-core-3.10.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-support-3.10.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ronn-0.7.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rexml-3.2.4/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rdiscount-2.2.0.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rack-2.2.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/pry-0.13.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/plist-3.6.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/patchelf-1.3.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rainbow-3.0.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/parallel_tests-3.4.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/parallel-1.20.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mustache-1.1.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/method_source-1.0.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mechanize-2.7.6/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/webrobots-0.1.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ntlm-http-0.1.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/net-http-persistent-4.0.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/net-http-digest_auth-1.4.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mime-types-3.3.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mime-types-data-3.2020.1104/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/http-cookie-1.0.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/hpricot-0.8.6/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/elftools-1.1.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/domain_name-0.5.20190701/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/unf-0.1.4/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/unf_ext-0.0.7.7/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/diff-lcs-1.4.4/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/connection_pool-2.2.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/commander-4.5.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/highline-2.0.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/colorize-0.8.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/coderay-1.1.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/simplecov-0.20.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/simplecov_json_formatter-0.1.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/simplecov-html-0.12.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/byebug-11.1.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/bindata-2.4.8/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ast-2.4.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/activesupport-6.1.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/zeitwerk-2.4.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/tzinfo-2.0.4/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/i18n-1.8.7/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/concurrent-ruby-1.1.7/lib/concurrent-ruby
/Library/Ruby/Site/2.6.0
/Library/Ruby/Site/2.6.0/universal-darwin20
/Library/Ruby/Site
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/vendor_ruby/2.6.0
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/vendor_ruby/2.6.0/universal-darwin20
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/vendor_ruby
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/universal-darwin20
/usr/local/Homebrew/Library/Homebrew
==> Homebrew Version
2.7.5-10-g8869208
==> macOS
11.1
==> SIP
Enabled
==> Java
15.0.1, 12.0.2, 1.8.0_66, 1.8.0_60
==> Homebrew Cask Staging Location
/usr/local/Caskroom
==> Homebrew Cask Taps:
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask (3794 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-versions (175 casks)
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask-drivers (170 casks)
```

#### Output of `brew tap`

```
heroku/brew
homebrew/cask
homebrew/cask-drivers
homebrew/cask-versions
homebrew/core
```

**Comment by miccal** [comment]
@todd-a-jacobs please run `brew update` and `brew upgrade fly` to confirm that https://github.com/Homebrew/homebrew-cask/pull/98460 has resolved your issue, thanks.

**Comment by todd-a-jacobs** [comment]
> please run `brew update` and `brew upgrade fly` to confirm that #98460 has resolved your issue, thanks.

@miccal I reinstalled the new cask, with the same results. I also tried:

``` bash
$ brew uninstall --force --cask fly

$ brew install --cask fly
==> Downloading https://github.com/concourse/concourse/releases/download/v6.7.3/
Already downloaded: /Users/foo/Library/Caches/Homebrew/downloads/a120d455c823c0f21e10436c38ef3dd63e06a1542ed77d2081723c8ee765d689--fly-6.7.3-darwin-amd64.tgz
==> Installing Cask fly
==> Linking Binary 'fly' to '/usr/local/bin/fly'.
🍺  fly was successfully installed!

$ xattr -l "$(brew --prefix)/bin/fly"
com.apple.quarantine: 0181;6009b76e;Homebrew\x20Cask;8141C7F1-1865-4CCA-B73E-B90BE948613E
```
So, the post-flight isn't working, but doing it manually fixes the issue:

``` bash
$ xattr -dr com.apple.quarantine "$(brew --prefix)/bin/fly"

# no output because quarantine is lifted
$ xattr -l "$(brew --prefix)/bin/fly"
```

**Comment by miccal** [comment]
@todd-a-jacobs I have merged https://github.com/Homebrew/homebrew-cask/pull/98494 -- please run `brew update` and `brew install --force fly` to confirm that this has resolved your issue, thanks.

**Comment by todd-a-jacobs** [comment]
> @todd-a-jacobs I have merged #98494 -- please run `brew update` and `brew install --force fly` to confirm that this has resolved your issue, thanks.

@miccal The new cask recipe worked! Thank you very much for getting this into the cask so quickly. Much appreciated!
